pzsd_bot/cogs/points.py

```diff
@@ -32,142 +36,153 @@ async def on_message(self, message: Message) -> None:
         if message.author == self.bot.user:
             return
 
+        recipient_name = recipient_id = None
+
         if match := POINT_PATTERN.search(message.content):
-            point_amount = int(match["point_amount"].replace(",", ""))
-            pretty_point_amount = format(point_amount, ",")
             recipient_name = match["recipient_name"]
             if recipient_name is not None:
                 recipient_name = recipient_name.strip('"')
             recipient_id = match["recipient_id"]
+        elif message.reference and (match := REPLY_POINT_PATTERN.fullmatch(message.content)):
+            original_message = self.bot.get_message(message.reference.message_id)
+            # message wasn't cached, make api call
+            if original_message is None:
+                original_message = await message.channel.fetch_message(message.reference.message_id)
+            recipient_id = str(original_message.author.id)
+        else:
+            return
 
-            async with Session.begin() as session:
+        point_amount = int(match["point_amount"].replace(",", ""))
+        pretty_point_amount = format(point_amount, ",")
+
+        async with Session.begin() as session:
```

Comment:
Everything from this point downward is unchanged aside from being de-dented once. Don't be fooled by the oddness of the diff.